### PR TITLE
many: have AuthContext expose device store-id, serial and serial-proof signing to the store

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: overlord/auth/auth.go [hunk 1]
```diff
@@ -208,21 +210,40 @@ NextUser:
 	return nil, ErrInvalidAuth
 }
 
-// An AuthContext handles user updates.
+// DeviceAssertions helps exposing the assertions about device indentity.
```
Comment (matiasb) [inline overlord/auth/auth.go]: s/indentity/identity

File: overlord/export_test.go [hunk 2]
```diff
@@ -53,3 +56,10 @@ func MockEnsureNext(o *Overlord, t time.Time) {
 func (o *Overlord) Engine() *StateEngine {
 	return o.stateEng
 }
+
+func MockStoreNew(new func(*store.Config, auth.AuthContext) *store.Store) (restore func()) {
```
Comment (matiasb) [inline overlord/export_test.go]: Should this have a doc comment?

File: store/store.go [hunk 3]
```diff
@@ -395,6 +399,21 @@ func (s *Store) authenticateDevice(r *http.Request) {
 	}
 }
 
+func (s *Store) setStoreID(r *http.Request) {
+	storeID := s.fallbackStoreID
+	if s.authContext != nil {
+		cand, err := s.authContext.StoreID(storeID)
```
Comment (matiasb) [inline store/store.go]: cand is from candidate? any better name?

**Comment by matiasb** [comment]
Changes makes sense to me.

File: overlord/export_test.go (hunk 2 above)
Comment (pedronis) [inline overlord/export_test.go]: we are quite inconsistent about giving doc comments to export_test.go stuff, but I can add one

File: store/store.go (hunk 3 above)
Comment (pedronis) [inline store/store.go]: yes, I tend to use cand quite a bit, it's storeIDCandidate but that's a bit long for a local variable for go

File: overlord/auth/auth.go [hunk 4]
```diff
@@ -248,3 +269,49 @@ func (ac *authContext) UpdateUser(user *UserState) error {
...
 	return UpdateUser(ac.state, user)
 }
+
+// StoreID returns the store id according to system state or
+// the fallback one if the state has none set (yet).
+func (ac *authContext) StoreID(fallback string) (string, error) {
+	if storeID := os.Getenv("UBUNTU_STORE_ID"); storeID != "" {
+		return storeID, nil
+	}
+	var storeID string
```
Comment (chipaca) [inline overlord/auth/auth.go]: strange that you'd do it this way and not move the getenv out of the if. worried about getenv leaving it dirty?

**Comment by chipaca** [comment]
:+1:

File: overlord/auth/auth.go (hunk 4 above)
Comment (pedronis) [inline overlord/auth/auth.go]: @chipaca just that this grew piecewise I think, I could do that now ... I'm not happy how complicated this one is actually, but we have the issue that in store we might have a authcontext or not (tool usage of store case)

**Pull request merged**